velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -731,6 +929,70 @@ static bool compareMaterializedRows(
   return true;
 }
 
+bool assertEqualResults(
+    const std::vector<RowVectorPtr>& expected,
+    const std::vector<RowVectorPtr>& actual) {
+  MaterializedRowMultiset expectedRows;
+  for (auto vector : expected) {
+    auto rows = materialize(vector);
+    std::copy(
+        rows.begin(),
+        rows.end(),
+        std::inserter(expectedRows, expectedRows.end()));
+  }
+
+  return assertEqualResults(expectedRows, actual);
+}
+
+// Compare actualRows with expectedRows and return whether they match. If
+// actualRows and expectedRows can be sorted by unique keys, compare their
+// values with epsilon. Otherwise, compare their values directly. The underlying
+// assumption is that aggregation results can be sorted by unique keys and
+// floating-point values in them are computed in different ways and hence
+// require epsilon comparison. For results of other operations, floating-point
+// values are likely copied from inputs and hence can be compared directly.
+bool assertEqualResults(
+    const MaterializedRowMultiset& expectedRows,
+    const MaterializedRowMultiset& actualRows,
+    const std::string& message) {
+  std::vector<MaterializedRow> actualReordered;
+  std::vector<MaterializedRow> expectedReordered;
+  bool haveUniqueKeys = sortByUniqueKey(
```

Comment:
haveUniqueKeys - this is a nice variable name, but it is misleading as sortByUniqueKey return true also when there are no floating-point columns. It might be cleaner to first identify the floating-point columns and then call sortByUniqueKey only if there is at least one such column.

Reply:
This is a good point, along with other good suggestions about this function. I'll update it. Thanks!